int32 pos
float64 cx
float64 area
bool appear
